update fleet yaml to reflect 1.1.0 spec reference

sumaithri opened the pull request:
updating the yaml to reflect the new APIs and fields added/modified for 1.1.0

VictoriaAuto commented on crd/fleet/fleet-crds.yaml:
```diff
@@ -3892,15 +5239,251 @@ spec:
+              newPrimary:
+                description: NewPrimary is the standby instance to promote as the
+                  new primary. If left empty, the system will automatically pick the
+                  best one to failover to.
```
What does "one" mean here? Instance? Replace "one" with the name of the thing itself to ensure user comprehension.

VictoriaAuto commented on crd/fleet/fleet-crds.yaml:
```diff
@@ -3892,15 +5239,251 @@ spec:
+            description: FailoverStatus represents the current state of a failover.
             properties:
               conditions:
                 description: Conditions represents the latest available observations
```
Add a period at the end of the sentence.

VictoriaAuto commented on crd/fleet/fleet-crds.yaml:
```diff
@@ -3975,6 +5558,11 @@ spec:
                 x-kubernetes-list-map-keys:
                 - type
                 x-kubernetes-list-type: map
+              createTime:
```
Replace "the time that the internal failover" with "the time when the internal failover"

VictoriaAuto commented on crd/fleet/fleet-crds.yaml:
```diff
@@ -4064,6 +5652,40 @@ spec:
                   type: object
                 type: array
+              endTime:
+                description: EndTime is the time failover reached its final state.
```
Add "when" after "time"

VictoriaAuto commented on crd/fleet/fleet-crds.yaml:
```diff
@@ -4064,6 +5652,40 @@ spec:
+                format: date-time
+                type: string
+              internal:
+                description: Internal is used by the system controllers. You should
```
Replace "You should not directly depend on..." with "Do not rely on the information in this section"

VictoriaAuto commented on crd/fleet/fleet-crds.yaml:
```diff
@@ -4064,6 +5652,40 @@ spec:
+                  not directly depend on the information in this section.
+                properties:
+                  newPrimary:
+                    description: NewPrimary is the instance that we are attempting
```
Replace "we" with "you" -- "we" is restricted for situations when we, as Google, recommend something to the user.

VictoriaAuto commented on crd/fleet/fleet-crds.yaml:
```diff
@@ -4064,6 +5652,40 @@ spec:
+                    type: string
+                  phase:
+                    description: Phase is used to keep track of the current state
+                      of the failover
```
Add a period at the end of the sentence.

VictoriaAuto commented on crd/fleet/fleet-crds.yaml:
```diff
@@ -4064,6 +5652,40 @@ spec:
+                - oldPrimary
+                type: object
               observedGeneration:
                 description: 'Internal: The generation observed by the controller.'
```
Remove quotation marks.

VictoriaAuto commented on crd/fleet/fleet-crds.yaml:
```diff
@@ -4072,11 +5694,37 @@ spec:
                 description: 'Internal: Whether the resource was reconciled by the
```
Remove because it's internal.

VictoriaAuto commented on crd/fleet/fleet-crds.yaml:
```diff
@@ -4072,11 +5694,37 @@ spec:
+              state:
+                description: State is the current state of the failover operation.
+                  The values are `InProgress`, `Success`, `Failed_RollbackInProgress`,
+                  `Failed_RollbackSuccess`, `Failed_RollbackFailed` `InProgress` means
```
- Add "and" after `Failed_RollbackSuccess`
- Add a period after `Failed_RollbackFailed`

VictoriaAuto commented on crd/fleet/fleet-crds.yaml:
```diff
@@ -4072,11 +5694,37 @@ spec:
+                  the failover is still in progress. `Success` means that the failover
+                  has completed. It is complete when the new primary instance is successfully
+                  promoted. `Failed_RollbackInProgress` means that the operator was
+                  unable to promote the new primary instance, and is attempting to
```
"and that it is attempting to"

VictoriaAuto commented on crd/fleet/fleet-crds.yaml:
```diff
@@ -4072,11 +5694,37 @@ spec:
+                  unable to promote the new primary instance, and is attempting to
+                  restart the old primary instance. `Failed_RollbackSuccess` means
+                  that the operator was unable to promote the new primary instance,
+                  and successfully restarted the old primary instance. `Failed_RollbackFailed`
```
"and that it successfully restarted"

VictoriaAuto commented on crd/fleet/fleet-crds.yaml:
```diff
@@ -4072,11 +5694,37 @@ spec:
+                  restart the old primary instance. `Failed_RollbackSuccess` means
+                  that the operator was unable to promote the new primary instance,
+                  and successfully restarted the old primary instance. `Failed_RollbackFailed`
+                  means that  the operator was unable to promote the new primary instance,
```
Remove extra space between "that" and "the"

VictoriaAuto commented on crd/fleet/fleet-crds.yaml:
```diff
@@ -4072,11 +5694,37 @@ spec:
+                  that the operator was unable to promote the new primary instance,
+                  and successfully restarted the old primary instance. `Failed_RollbackFailed`
+                  means that  the operator was unable to promote the new primary instance,
+                  and were not able to restart the old primary instance. The DBCluster
```
Replace "were not able to restart" with "and that the operator wasn't able to restart..."

VictoriaAuto commented on crd/fleet/fleet-crds.yaml:
```diff
@@ -4085,28 +5733,36 @@ metadata:
       openAPIV3Schema:
-        description: Failover represents the parameters and status of a single failover
-          operation.
+        description: Replication is the Schema for the Replications API
```
Add a period at the end of the sentence.

VictoriaAuto commented on crd/fleet/fleet-crds.yaml:
```diff
@@ -4121,22 +5777,103 @@ spec:
           spec:
-            description: FailoverSpec represents the parameters of a single failover
-              operation.
+            description: ReplicationSpec defines the desired state of Replication
```
Add a period at the end of the sentence.

VictoriaAuto commented on crd/fleet/fleet-crds.yaml:
```diff
@@ -4121,22 +5777,103 @@ spec:
-                  best one to failover to.
-                type: string
+              dbcluster:
+                description: DBCluster is the DBCluster this Replication belongs to.
```
Add "that" between "DBCluster" and "this Replication"

VictoriaAuto commented on crd/fleet/fleet-crds.yaml:
```diff
@@ -4121,22 +5777,103 @@ spec:
+                description: DBCluster is the DBCluster this Replication belongs to.
+                properties:
+                  name:
+                    description: 'Name of the referent. More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names
```
- Remove quotation marks from the beginning and the end of the sentence.
- Remove the TODO comment.

VictoriaAuto commented on crd/fleet/fleet-crds.yaml:
```diff
@@ -4121,22 +5777,103 @@ spec:
+                type: object
+                x-kubernetes-map-type: atomic
+              downstream:
+                description: Downstream contains spec for replication downstream database.
```
Replace "spec" with "specifications"?

VictoriaAuto commented on crd/fleet/fleet-crds.yaml:
```diff
@@ -4121,22 +5777,103 @@ spec:
+                  password:
+                    description: Password is the reference to the secret storing replication
+                      user password. If not provided, a password will be generated
+                      and stored in a secret shown in status.
```
Replace "will be" with "is generated"

VictoriaAuto commented on crd/fleet/fleet-crds.yaml:
```diff
@@ -4121,22 +5777,103 @@ spec:
+                  replicationSlotName:
+                    description: ReplicationSlotName is the name of the replication
+                      slot to be used for replication. If not provided, a replication
+                      slot name will be generated and shown in status.
```
Replace "will be generated" with "is generated"

VictoriaAuto commented on crd/fleet/fleet-crds.yaml:
```diff
@@ -4121,22 +5777,103 @@ spec:
+                    type: string
+                  username:
+                    description: Username is the name of the replication user to be
+                      used for replication. If not provided, a username will be generated
```
Replace "will be generated" with "is generated"

VictoriaAuto commented on crd/fleet/fleet-crds.yaml:
```diff
@@ -4121,22 +5777,103 @@ spec:
             type: object
           status:
-            description: FailoverStatus represents the current state of a failover.
+            description: ReplicationStatus defines the observed state of Replication
```
Add a period at the end of the sentence.

VictoriaAuto commented on crd/fleet/fleet-crds.yaml:
```diff
@@ -7765,9 +9488,14 @@ spec:
               newPrimaryInstance:
-                description: NewPrimaryInstance is the standby instance to switch
-                  with the current primary.
+                description: 'NewPrimaryInstance is the standby instance to switch
```
Quotation marks from the beginning and end of the sentence.

sumaithri replied:
do I need to update this to double quotes?

VictoriaAuto replied:
Yes. Thank you!

VictoriaAuto reviewed (commented):
Remove "accordingly

sumaithri commented on crd/fleet/fleet-crds.yaml:
```diff
@@ -1569,93 +1640,739 @@ spec:
+                              preferred is the one with the greatest sum of weights,
+                              i.e. for each node that meets all of the scheduling
+                              requirements (resource request, requiredDuringScheduling
+                              affinity expressions, etc.), compute a sum by iterating
```
the scheduler does these calculations and picks the node that has the highest sum of weights.
description starts with the line "The scheduler will prefer..."

sumaithri replied:
how about this

```
description: The scheduler will prioritize nodes that meet the affinity
                              expressions specified in this field when scheduling pods,
                              but it may still choose a node that doesn't fully meet the
                              criteria. The most preferred node is the one with the highest
                              total weight. This weight is calculated by adding up values
                              for each node that satisfies all scheduling requirements
                              (such as resource requests or requiredDuringScheduling affinity
                              expressions). For each matching podAffinityTerm, the node's weight
                              increases, and the node(s) with the highest total weight are
                              selected as the most preferred.
```

VictoriaAuto replied:
Please replace "may" with "might" -- otherwise, LGTM! Thank you.

sumaithri commented on crd/fleet/fleet-crds.yaml:
```diff
@@ -1569,93 +1640,739 @@ spec:
+                                      type: object
+                                      x-kubernetes-map-type: atomic
+                                    namespaceSelector:
+                                      description: A label query over the set of namespaces
```
over seems better - as it is referring to a list of namespaces that are available

VictoriaAuto replied:
LGTM!

sumaithri commented:
@VictoriaAuto  - this PR was auto-merged by the admin. I have addressed your comments in another PR and left a couple of questions here
https://github.com/GoogleCloudPlatform/alloydb-omni-samples/pull/10
(once this all looks good - I'll update it in our internal codebase as well)

VictoriaAuto commented on crd/fleet/fleet-crds.yaml:
```diff
@@ -1569,93 +1640,739 @@ spec:
-                              TODO: Add other useful fields. apiVersion, kind, uid?'
-                            type: string
+                          preferredDuringSchedulingIgnoredDuringExecution:
+                            description: The scheduler will prefer to schedule pods
```
Yes. I'm referring to the sentence with "The scheduler will prefer to schedule pods..." Please replace "will prefer" with "prefers" -- Cloud iX guidance is to use the present tense whenever possible. Thank you!

sumaithri commented on crd/fleet/fleet-crds.yaml:
```diff
@@ -1569,93 +1640,739 @@ spec:
+                          rules for the instance.
+                        properties:
+                          preferredDuringSchedulingIgnoredDuringExecution:
+                            description: The scheduler will prefer to schedule pods
```
done - replacing it with `prefers`

sumaithri commented on crd/fleet/fleet-crds.yaml:
```diff
@@ -1569,93 +1640,739 @@ spec:
+                    additionalProperties:
+                      type: boolean
+                    description: Services list the optional semi-managed services
+                      that the customers can choose from.
```
good one!

sumaithri commented on crd/fleet/fleet-crds.yaml:
```diff
@@ -1569,93 +1640,739 @@ spec:
+                    properties:
+                      name:
+                        description: 'Name of the referent. More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names
+                          TODO: Add other useful fields. apiVersion, kind, uid?'
```
removed it from about 25 places in the file

sumaithri commented on crd/fleet/fleet-crds.yaml:
```diff
@@ -3562,34 +4293,650 @@ spec:
+                              properties:
+                                labelSelector:
+                                  description: A label query over a set of resources,
+                                    in this case pods.
```
yep done, all 8 occurrences replaced

sumaithri commented on crd/fleet/fleet-crds.yaml:
```diff
@@ -3562,34 +4293,650 @@ spec:
+                        type: array
+                      requiredDuringSchedulingIgnoredDuringExecution:
+                        description: If the anti-affinity requirements specified by
+                          this field are not met at scheduling time, the pod will
```
resolved in 6 other locations

sumaithri commented on crd/fleet/fleet-crds.yaml:
```diff
@@ -3562,34 +4293,650 @@ spec:
+                          not be scheduled onto the node. If the anti-affinity requirements
+                          specified by this field cease to be met at some point during
+                          pod execution (e.g. due to a pod label update), the system
+                          may or may not try to eventually evict the pod from its
```
all replaced!

sumaithri commented on crd/fleet/fleet-crds.yaml:
```diff
@@ -3562,34 +4293,650 @@ spec:
+                          pod execution (e.g. due to a pod label update), the system
+                          may or may not try to eventually evict the pod from its
+                          node. When there are multiple elements, the lists of nodes
+                          corresponding to each podAffinityTerm are intersected, i.e.
```
replaced 3 instances of it

sumaithri commented on crd/fleet/fleet-crds.yaml:
```diff
@@ -3562,34 +4293,650 @@ spec:
+                                      operator:
+                                        description: operator represents a key's relationship
+                                          to a set of values. Valid operators are
+                                          In, NotIn, Exists and DoesNotExist.
```
replaced all 15 instances

sumaithri commented on crd/fleet/fleet-crds.yaml:
```diff
@@ -3562,34 +4293,650 @@ spec:
+                                union of the namespaces selected by this field and
+                                the ones listed in the namespaces field. null selector
+                                and null or empty namespaces list means "this pod's
+                                namespace". An empty selector ({}) matches all namespaces.
```
checked for most of these occurrences and they are all either actual literal values or error codes.
as per link, it seems to fall under acceptable
 `Acceptable: If you enter "escape", the program crashes.`

sumaithri commented on crd/fleet/fleet-crds.yaml:
```diff
@@ -1569,93 +1640,739 @@ spec:
+                              requirements (resource request, requiredDuringScheduling
+                              anti-affinity expressions, etc.), compute a sum by iterating
+                              through the elements of this field and adding "weight"
+                              to the sum if the node has pods which matches the corresponding
```
replaced all 5 occurrences

sumaithri commented on crd/fleet/fleet-crds.yaml:
```diff
@@ -3892,15 +5239,251 @@ spec:
+                description: Conditions represents the latest available observations
+                  of the Entity's current state.
+                items:
+                  description: "Condition contains details for one aspect of the current
```
removed all 15 occurrences

sumaithri commented on crd/fleet/fleet-crds.yaml:
```diff
@@ -3892,15 +5239,251 @@ spec:
+                  properties:
+                    lastTransitionTime:
+                      description: lastTransitionTime is the last time the condition
+                        transitioned from one status to another. This should be when
```
changed it all 13 places

sumaithri commented on crd/fleet/fleet-crds.yaml:
```diff
@@ -3892,15 +5239,251 @@ spec:
+                        --- Many .condition.type values are consistent across resources
+                        like Available, but because arbitrary conditions can be useful
+                        (see .node.status.conditions), the ability to deconflict is
+                        important. The regex it matches is (dns1123SubdomainFmt/)?(qualifiedNameFmt)
```
all 15 occurrences

sumaithri commented on crd/fleet/fleet-crds.yaml:
```diff
@@ -3892,15 +5239,251 @@ spec:
+                    resource:
+                      description: Resource contains information about the Database
+                        Service component that reported the incident as well as about
+                        the K8s resource.
```
yes! replaced about 100 occurrences of k8s, K8s, K8S to Kubernetes
